samples/06-prs/provider/src/main/java/org/eclipse/dataspaceconnector/runtime/ConnectorRuntime.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2020, 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.runtime;
+
+import org.eclipse.dataspaceconnector.monitor.MonitorProvider;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.system.DefaultServiceExtensionContext;
+
+import java.util.List;
+import java.util.ListIterator;
+
+import static org.eclipse.dataspaceconnector.system.ExtensionLoader.bootServiceExtensions;
+import static org.eclipse.dataspaceconnector.system.ExtensionLoader.loadMonitor;
+import static org.eclipse.dataspaceconnector.system.ExtensionLoader.loadVault;
+
+/**
+ * Main entrypoint for the default runtime.
+ * Custom runtime is added to be able to debug the provider.
```

Comment:
As discussed searched and found a way to debug if main entry point is not defined in gradle application.
Maybe we can test if it works without the ConnectorRuntime for now.